processing/src/main/java/org/apache/druid/query/filter/InDimFilter.java

```diff
@@ -132,31 +145,40 @@ public FilterTuning getFilterTuning()
   @Override
   public byte[] getCacheKey()
   {
-    boolean hasNull = false;
-    for (String value : values) {
-      if (value == null) {
-        hasNull = true;
-        break;
+    if (cacheKey == null) {
+      final List<String> sortedValues = new ArrayList<>(values);
```

Comment:
Does this negate your comment on line 96?

We'd be copying all the values here to a list so that we can sort the list.

Reply:
This method is not called if populateCache = false and useCache = false. Will add some comments about it.

Reply:
Added.